Infrastructure/Modules/CDNHelpers.psm1

```diff
@@ -0,0 +1,204 @@
+        [Parameter(Mandatory = $true)]
+        [String]$SaSToken,
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("Storage", "Cloud Service", "Web App", "Custom Origin")]
+        [String]$OriginType
+    )
+    try {
+        if ($OriginType -eq "Storage") {
+            # --- Set location for AzCopy.exe
+            $AzCopyPath = 'C:\Program Files (x86)\Microsoft SDKs\Azure\AzCopy\'
+            if (Test-Path $AzCopyPath ) {
+                Write-Log -LogLevel Information -Message "Setting location to Azure Storage AzCopy utility"
+                Set-Location -Path $AzCopyPath
+            }
+            else {
+                Write-Log -LogLevel Information -Message "Could not locate Azure Storage AzCopy utility under $AzCopyPath"
+            }
+            Write-Log -LogLevel Information -Message "Invoking Azure Storage AzCopy utility to upload content and change MIME settings"
+            # ---> Invoke AzCopy.exe for
+            .\AzCopy.exe /Source:$Source /Dest:$Destination /DestKey:$SaStoken /NC:10 /Z /V /S /Y /SetContentType
+            # ---> Invoke AzCopy.exe for *.woff
+            .\AzCopy.exe /Source:$Source /Dest:$Destination /DestKey:$SaStoken /NC:10 /Z /V /Pattern:"*.woff" /SetContentType:"application/font-woff" /S /Y
+            # ---> Invoke AzCopy.exe for *.woff2
+            .\AzCopy.exe /Source:$Source /Dest:$Destination /DestKey:$SaStoken /NC:10 /Z /V /Pattern:"*.woff2" /SetContentType:"application/font-woff2" /S /Y
+        }
+        else {
+            Write-Log -LogLevel Information -Message "Blob copy not required as OriginType set to either 'Cloud Service', 'Web App' or 'Custom Origin'"
+        }
+    }
+    catch {
+        throw "Failed to copy content to blob and set MIME settings: $_"
+    }
+}
+
+function Enable-CORS {
+    <#
+.SYNOPSIS
+Set CORS settings on blob storage
+
+.DESCRIPTION
+Cross-Origin Resource Sharing (CORS) is a mechanism that uses additional HTTP headers to tell a browser
+to let a web application running at one origin (domain) have permission to access selected resources from a server at a different origin.
+Courtesy: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+More Information: https://docs.microsoft.com/en-us/powershell/module/azure.storage/set-azurestoragecorsrule?view=azurermps-6.12.0
+
+.PARAMETER StorageAccountName
+The StorageAccountName to apply the CORS settings
+
+.PARAMETER SaSToken
+The SaS Token to access the blob storage container
+
+.PARAMETER EnableCORS
+The CORS Rules to be applied
+
+.EXAMPLE
+
+$DeploymentParameters = @ {
+    StorageAccountName = "mystorageaccountname"
+    SaSToken = "MySecureSaStokenString"
+    $EnableCORS = "Yes"
+}
+Enable-CORS @DeploymentParameters
+
+#>
+
+    Param(
+        [Parameter(Mandatory = $true)]
+        [String]$StorageAccountName,
+        [Parameter(Mandatory = $true)]
+        [String]$SaSToken,
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("Yes", "No")]
+        [string]$EnableCORS
```

Comment:
The function is called "Enable-CORS", we shouldn't need a parameter with the same name. If this function is being called then it can be assumed that the caller wishes to enable CORS.